ethtx/providers/node/pool.py

```diff
@@ -0,0 +1,45 @@
+#  Copyright 2021 DAI Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Dict, List
+
+from .base import NodeBase
+
+
+class NodeConnectionPool:
+    def __init__(self, nodes: Dict[str, dict]):
+        self._set_connections(nodes)
+
+    def __getattribute__(self, chain: str) -> List[NodeBase]:
+        return super().__getattribute__(chain.lower())
+
+    def __setattr__(self, chain: str, value: NodeBase) -> None:
+        if not isinstance(value, NodeBase):
+            raise ValueError("Value is not instance of NodeBase")
+
+        if chain in self.__dict__:
+            self.__dict__[chain.lower()].append(value)
+        else:
+            self.__dict__[chain.lower()] = [value]
+
+    def __len__(self) -> int:
+        return len(self.__dict__)
+
+    @property
+    def connections(self) -> Dict:
+        return self.__dict__
+
+    def _set_connections(self, nodes) -> None:
+        for chain, node_params in nodes.items():
+            nodes: List[str] = list(node_params.values())[0].split(",")
+            poa: bool = list(node_params.values())[1]
+            for node in nodes:
+                self.__setattr__(chain, NodeBase(node=node, poa=poa))
```

Comment:
I don't see any advantage of using build-in `__dict__` instead of normal dict for holding connections, but there disadvantages. Without reading this class code it's impossible to use it, since the only thing exposed is `connections` field. Instead of `__setattr__` there should be `add_connections`, for `__getattribute__` is should be `get_connections`, making it easier to use and understand. Also, now it's pretty much impossible to extend. Please change it. Second thing, using `lower`, there is no reason why chain_id should be case insensitive, an could cause bugs. It should be users choice if he wants to use lower case chain_ids